src/main/java/unicash/logic/parser/FindCommandParser.java

```diff
@@ -1,36 +1,127 @@
+        if (trimmedArgs.isEmpty() || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_TYPE, PREFIX_AMOUNT,
+                PREFIX_DATETIME, PREFIX_CATEGORY, PREFIX_LOCATION);
+
+        /* Parses the name keywords provided and adds them to the all keywords predicate */
+        for (String keyword : argMultimap.getAllValues(PREFIX_NAME)) {
+            Name transactionName = ParserUtil.parseTransactionName(keyword);
+            findPredicate.addNameKeyword(transactionName.toString());
+        }
+
+        /* Parses the amount keywords provided and adds them to the all keywords predicate */
+        for (String keyword : argMultimap.getAllValues(PREFIX_AMOUNT)) {
+            Amount transactionAmount = ParserUtil.parseAmount(keyword);
+            findPredicate.addAmountKeyword(
+                    Amount.amountToDecimalString(transactionAmount));
+        }
+
+        /* Parses the categories provided and adds them to the all keywords predicate */
+        for (String keyword : argMultimap.getAllValues(PREFIX_CATEGORY)) {
+            Category transactionCategory = ParserUtil.parseCategory(keyword);
+            findPredicate.addCategoryKeyword(transactionCategory.toString());
+        }
+
+        /* Parses the location keywords provided and adds them to the all keywords predicate */
+        for (String keyword : argMultimap.getAllValues(PREFIX_LOCATION)) {
+            Location transactionLocation = ParserUtil.parseLocation(keyword);
+            findPredicate.addLocationKeyword(transactionLocation.toString());
+        }
+
+        /* Parses the dateTime keywords provided and adds them to the all keywords predicate */
+        for (String keyword : argMultimap.getAllValues(PREFIX_DATETIME)) {
+            DateTime transactionDateTime = ParserUtil.parseDateTime(keyword);
+            findPredicate.addDateTimeKeyword(transactionDateTime.toString());
+        }
+
+        /* Parses the type keyword provided and adds it to the all keywords predicate */
+        for (String keyword : argMultimap.getAllValues(PREFIX_TYPE)) {
+            Type transactionType = ParserUtil.parseType(keyword);
+            findPredicate.addTypeKeyword(transactionType.toString());
+        }
+
+        return new FindCommand(findPredicate);
 
-        return new FindCommand(new TransactionContainsKeywordsPredicate(
-                Arrays.asList(nameKeywords)));
     }
 
 
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof FindCommandParser)) {
+            return false;
+        }
+
+        FindCommandParser otherCommand = (FindCommandParser) other;
```

Comment:
I believe traditionally, the first condition is to check if `other == null` and return `false` directly then

Reply:
Fixed!